### PR TITLE
postgresqlPackages.pgvecto-rs: init at 0.2.1

### PR DESCRIPTION
**Pull request by esclear**
## Description of changes

From the [pgvecto.rs repository](https://github.com/tensorchord/pgvecto.rs):
> pgvecto.rs is a Postgres extension that provides vector similarity search functions. It is written in Rust and based on [pgrx](https://github.com/tcdi/pgrx).

Big thanks to @katrinafyi, who helped me getting the clang build going.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sikmir** [comment]
Commit message `pgvectors: init at 0.2.0`, please.

File: pkgs/servers/sql/postgresql/ext/pgvectors/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  buildInputs = [
+    postgresql
+    openssl
+  ];
+
+  meta = with lib; {
+    description = "Scalable Vector database plugin for Postgres, written in Rust, specifically designed for LLM";
+    homepage = "https://github.com/tensorchord/pgvecto.rs";
+    maintainers = with maintainers; [ esclear ];
+    platforms = [ "x86_64-linux" ];
```
Comment (sikmir) [inline pkgs/servers/sql/postgresql/ext/pgvectors/default.nix:59]:
```suggestion
    platforms = platforms.linux;
```

**Comment by SuperSandro2000** [comment]
@ofborg build postgresqlPackages.pgvectors

**Comment by SuperSandro2000** [comment]
Fails to build on aarch64-linux, see ofborg log

**Comment by marsam** [comment]
Hi, sorry for the delay. It looks great, thank you!   I'll take a proper look tonight

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests = {
+      pgvecto-rs = nixosTests.pgvecto-rs;
+    };
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
```
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:69]: Clang version isn't locked here like the other PR does.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ];
+  buildInputs = [
+    postgresql
+    openssl
+  ];
+
+  meta = with lib; {
+    description = "Scalable Vector database plugin for Postgres, written in Rust, specifically designed for LLM";
+    homepage = "https://github.com/tensorchord/pgvecto.rs";
+    maintainers = with maintainers; [ esclear ];
```
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:80]: @diogotcorreia would you like to be added as a maintainer here?

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    platforms = [ "x86_64-linux" ];
```
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:81]: From the other PR:

```suggestion
    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
```

**Pull request closed without merging**